awx/main/models/events.py

```diff
@@ -579,6 +582,13 @@ def _update_host_summary_from_stats(self, hostnames):
                     h.last_job_host_summary_id = host_mapping[h.id]
                     updated_hosts.add(h)
 
+                # Populate unique host retention table if enabled
+                if settings.UNIQUE_HOST_RETENTION_ENABLED:
+                    # If host already exists, update it, else create it
+                    UniqueHost.objects.update_or_create(
```

Comment:
This is really critical part of the application in terms of performance. If you see below, we do a `bulk_update` because we found individual insertions to be problematic.